File: python/arithmetic-subarrays.py
class Solution:
    def is_aritmetic_subarray(nums_subarray: List[int]) -> bool:
        if len(nums_subarray) < 2:
            return False
        nums_subarray.sort()
        diff = nums_subarray[1] - nums_subarray[0]
        for i in range(1, len(nums_subarray)):
            if (nums_subarray[i] - nums_subarray[i - 1]) != diff:
                return False
        return True

    def checkArithmeticSubarrays(self, nums: List[int], l: List[int], r: List[int]) -> List[bool]:
        output = []
        for i in range(len(l)):
            start, end = l[i], r[i] + 1
            output.append(Solution.is_aritmetic_subarray(nums[start:end]))
        return output

File: go/arithmetic-subarrays.go
func IsArithmeticSubarray(subarray []int) bool {
	if len(subarray) < 2 {
		return false
	}
	sort.Ints(subarray)
	diff := subarray[1] - subarray[0]
	for i := 1; i < len(subarray); i++ {
		if (subarray[i] - subarray[i-1]) != diff {
			return false
		}
	}
	return true
}

func checkArithmeticSubarrays(nums []int, l []int, r []int) []bool {
	output := make([]bool, 0)
	for i := 0; i < len(l); i++ {
		start := l[i]
		end := r[i] + 1
		subarray := make([]int, end-start)
		copy(subarray, nums[start:end])
		output = append(output, IsArithmeticSubarray(subarray))
	}
	return output
}
